internal/format/asciidoc_table.go

```diff
@@ -0,0 +1,163 @@
+package format
+
+import (
+	"text/template"
+
+	"github.com/segmentio/terraform-docs/pkg/print"
+	"github.com/segmentio/terraform-docs/pkg/tfconf"
+	"github.com/segmentio/terraform-docs/pkg/tmpl"
+)
+
+const (
+	asciidocTableHeaderTpl = `
+	{{- if .Settings.ShowHeader -}}
+		{{- with .Module.Header -}}
+			{{ sanitizeHeader . }}
+			{{ printf "\n" }}
+		{{- end -}}
+	{{ end -}}
+	`
+
+	asciidocTableRequirementsTpl = `
+	{{- if .Settings.ShowRequirements -}}
+		{{ indent 0 "=" }} Requirements
+		{{ if not .Module.Requirements }}
+			No requirements.
+		{{ else }}
+			[cols="a,a",options="header,autowidth"]
+			|===
+			|Name |Version
+			{{- range .Module.Requirements }}
+				|{{ .Name }} |{{ tostring .Version | default "n/a" }}
+			{{- end }}
+			|===
+		{{ end }}
+	{{ end -}}
+	`
+
+	asciidocTableProvidersTpl = `
+	{{- if .Settings.ShowProviders -}}
+		{{ indent 0 "=" }} Providers
+		{{ if not .Module.Providers }}
+			No provider.
+		{{ else }}
+			[cols="a,a",options="header,autowidth"]
+			|===
+			|Name |Version
+			{{- range .Module.Providers }}
+				|{{ .FullName }} |{{ tostring .Version | default "n/a" }}
+			{{- end }}
+			|===
+		{{ end }}
+	{{ end -}}
+	`
+
+	asciidocTableInputsTpl = `
+	{{- if .Settings.ShowInputs -}}
+		{{ indent 0 "=" }} Inputs
+		{{ if not .Module.Inputs }}
+			No input.
+		{{ else }}
+			[cols="a,a,a,a{{ if .Settings.ShowRequired }},a{{ end }}",options="header,autowidth"]
+			|===
+			|Name |Description |Type |Default {{ if .Settings.ShowRequired }}|Required {{ end }}
```

Comment:
Very minor comment, I'm not sure how trailing space at the end of line would work in AsciiDoc, but this line will have an extra space after `|Default<extra-space>` if `ShowRequired` is false. May be change it to something like this, if needed?
```
|Name |Description |Type |Default{{ if .Settings.ShowRequired }} |Required {{ end }}
```

Reply:
You are right, trailing whitespace at the end of table rows is unnecesary. I've made some changes.

Reply:
Awesome, thanks.